packages/compass-schema/src/components/compass-schema.tsx

```diff
@@ -0,0 +1,456 @@
+import React, { useCallback, useEffect } from 'react';
+
+import type { AnalysisState } from '../constants/analysis-states';
+import {
+  ANALYSIS_STATE_INITIAL,
+  ANALYSIS_STATE_ANALYZING,
+  ANALYSIS_STATE_COMPLETE,
+} from '../constants/analysis-states';
+
+import { SchemaToolbar } from './schema-toolbar/schema-toolbar';
+import Field from './field';
+import { ZeroGraphic } from './zero-graphic';
+
+import {
+  Button,
+  CancelLoader,
+  css,
+  cx,
+  DocumentIcon,
+  EmptyContent,
+  Link,
+  palette,
+  spacing,
+  useDarkMode,
+  WorkspaceContainer,
+  lighten,
+  rgba,
+} from '@mongodb-js/compass-components';
+
+const mcBlue0 = palette.blue.light1;
+const mcBlue1 = lighten(0.075, mcBlue0);
+const mcBlue2 = lighten(0.15, mcBlue0);
+const mcBlue3 = lighten(0.225, mcBlue0);
+const mcBlue4 = lighten(0.3, mcBlue0);
+const mcBlue5 = lighten(0.375, mcBlue0);
+const mcBg = palette.gray.light2;
+const mcFg = mcBlue0;
+const mcFgSelected = palette.yellow.base;
+const mcFgUnselected = mcBg;
+
+const rootStyles = css`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  flex-grow: 1;
+  flex-shrink: 1;
+`;
+
+const loaderStyles = css`
+  height: 100%;
+  display: flex;
+  justify-content: center;
+`;
+
+const schemaStyles = css`
+  width: 100%;
+  padding: 0 ${spacing[3]}px;
+  flex-grow: 1;
+  overflow: auto;
+`;
+
+const minichartStyles = (darkMode: boolean) => css`
```

Comment:
There seems to be only a few styles here that switch by `darkMode`, as we are not refactoring this in any way, would it make more sense to just use injectGlobal and just put dark mode specific stuff behind the data-theme right in css. I'm thinking it might make it a bit clearer that the work here is not finished for the future changes. And as I mentioned in another comment separating stuff that changes based on dark mode from main styles just makes it easier to refactor in the future, it was quite hard to read though all the code here to see what exactly is switched based on the dark more

Reply:
it totally makes sense, yeah, it didn't come to mind as an option. It is not making a big difference I'd keep it as it is for now, I'd really like to move them where they are used.